android/src/main/kotlin/com/qonversion/flutter/sdk/qonversion_flutter_sdk/QonversionFlutterSdkPlugin.kt

```diff
@@ -111,14 +109,11 @@ class QonversionFlutterSdkPlugin internal constructor(registrar: Registrar): Met
         result.success(null)
     }
 
-    private fun createSkuDetails(map: Map<String, Any>): SkuDetails {
-        val json = map.toString()
+    private fun createSkuDetails(json: String): SkuDetails {
```

Comment:
Looks like this two functions can be removed and you can use `SkuDetails(json)` directly